safe-global/safe-react
# fix: Crash if spending limit module is not deployed on a chain

## What it solves
Resolves #3889 

## How this PR fixes it

- `getAllowanceModuleDeployment` can't throw. It instead returns `undefined` if the module is not deployed on a network. Instead of manually throwing on our side, we handle the `undefined` value.
- `getSpendingLimitModuleAddress` can return `string | undefined` instead of just `string`

## How to test it

1. Open one of these Safes: `arb1:0x2EBA2F2F1D599E9E2920ce4B70b1B881e76db82E`, `aurora:0xDBe6f953a53eE0Eb7D28101e123D7bf6553638eb`
2. Observe the app not crashing
